plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs

```diff
@@ -166,13 +177,27 @@ prop_agree_Ck_Cek tyG tmG = do
   tm <- withExceptT GenError $ convertClosedTerm tynames names tyG tmG
   withExceptT TypeError $ checkType tcConfig () tm (Normalized ty)
 
-  -- check if CK and CEK give the same output
   tmCek <- withExceptT CekP $ liftEither $
     evaluateCek testBuiltinsRuntime tm `catchError` handleError ty
 
   tmCk <- withExceptT CkP $ liftEither $
     evaluateCk testBuiltinsRuntime tm `catchError` handleError ty
-  unless (tmCk == tmCek) $ throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
+
+  -- check if CK and CEK give the same output
+  unless (tmCk == tmCek) $
+    throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
```

Comment:
Do they? The typed CEK machine does not handle types correctly while the CK machine does. Please remind me how this discrepancy is resolved in NEAT tests.

Reply:
It only produces something at type unit which has no type annotations.

Reply:
I see. Is that documented?

Reply:
Good point. I have added some comments and removed an misleading one.